util/spatial-groups/spatial-groups.lisp

```diff
@@ -0,0 +1,153 @@
+;; Spatial Groups for StumpWM
+;;
+;; Copyright (C) 2022 Russell Adams
+;;
+;; Maintainer: Russell Adams <rladams@adamsinfoserv.com>
+;;
+
+;; This module is free software: you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 2 of the License, or
+;; (at your option) any later version.
+;;
+;; This module is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+;;
+;; You should have received a copy of the GNU General Public License
+;; along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Usage:
+;;
+;; Just add the following line to your .stumpwmrc file:
+;;
+;; (load-module :stumpwm-spatial)
+;;
+
+(in-package :spatial-groups)
+
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Define global vars
+
+(defparameter *z-cursors* '()
+  "Association list of groups by Z coordinate for preserving last position on that plane.")
+
+(defparameter *last-cursor* NIL
+  "Previous group name string for popping back.")
+
+(defparameter *spatial-banish-on-move* t
+  "Banish the cursor to bottom right when switching screens? Defaults to true.")
+
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Coordinate commands
+
+;; Control arrows between groups
+(defcommand coord-left      () () (coord-group-change -1  0  0))
+(defcommand coord-right     () () (coord-group-change  1  0  0))
+(defcommand coord-up        () () (coord-group-change  0  1  0))
+(defcommand coord-down      () () (coord-group-change  0 -1  0))
+(defcommand coord-taskleft  () () (coord-group-change  0  0 -1))
+(defcommand coord-taskright () () (coord-group-change  0  0  1))
+
+;; return to 0,0 on the current taskplane as a shortcut to return to the core task
+(defcommand coord-taskorigin () ()
+  (gselect
+   (group-name
+    (spatial-find-group (current-screen)
+                   (format nil "~{~a~^,~}" (list 0 0 (parse-integer (third (split-string (group-name (current-group)) ",")))))))))
+
+;; pop back to last location
+(defcommand coord-taskpop () () (when *last-cursor* (spatial-gselect *last-cursor*)))
+
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Key bindings for spatial group movement
+
+(defun install-default-keybinds ()
```

Comment:
I think it'd be nifty (in general) if we didn't just inject stuff into people's top-map, and just accepted the map as an argument so people could put a prefix (if they wanted) or just be more clear where things were going. 

 I'm less familar with the idiom inside of the contrib modules (my config has too much copy-and-paste from the olden days, and this seems like an ok thing to do in terms of expectations.